### PR TITLE
Fixed CMake Flow for Building Sparta

### PR DESCRIPTION
**Pull request by kathlenemagnus**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -63,30 +63,39 @@ set(SPARTA_CXX_FLAGS_DEBUG    "${CMAKE_CXX_FLAGS_DEBUG}")
...
+  elseif(DEFINED ENV{CONDA_PREFIX})
+    message (STATUS "Looking for SPARTA in the conda environment: '$ENV{CONDA_PREFIX}'")
+    set(SPARTA_SEARCH_DIR $ENV{CONDA_PREFIX})
+  else()
+    message (STATUS "If needed, please provide the location where sparta is installed: -DSPARTA_SEARCH_DIR=<directory>")
+  endif()
+  set(CMAKE_MODULE_PATH "${SPARTA_SEARCH_DIR}/lib/cmake/sparta" ${CMAKE_MODULE_PATH})
+
+  find_package(Sparta REQUIRED)
+  set(SPARTA "SPARTA::sparta")
```
Comment (colby-nyce) [inline CMakeLists.txt]: set(SPARTA_LIBS "SPARTA::sparta")

File: CMakeLists.txt [hunk 2]
```diff
@@ -63,30 +63,39 @@ set(SPARTA_CXX_FLAGS_DEBUG    "${CMAKE_CXX_FLAGS_DEBUG}")
...
-  message (STATUS "Looking for SPARTA in the conda environment: '$ENV{CONDA_PREFIX}'")
-  set(SPARTA_SEARCH_DIR $ENV{CONDA_PREFIX})
+if(BUILD_SPARTA_FROM_SOURCE)
+  message (STATUS "Building SPARTA from source in ${SPARTA_SEARCH_DIR}")
+  add_subdirectory(${SPARTA_SEARCH_DIR} "${CMAKE_BINARY_DIR}/sparta")
+  include_directories(${SPARTA_SEARCH_DIR})
+  include(${SPARTA_SEARCH_DIR}/cmake/SpartaTestingMacros.cmake)
+  include(${SPARTA_SEARCH_DIR}/cmake/sparta-config.cmake)
+  set(SPARTA_LIBS ${Sparta_LIBS})
+  set(SPARTA "${Sparta_LIBS}")
```
Comment (colby-nyce) [inline CMakeLists.txt]: Can this be removed?

File: test/actions/CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 project(Actions_Test)
 
 add_executable(Actions_test Actions_test.cpp)
-target_link_libraries(Actions_test atlascore atlasinsts SPARTA::sparta mavis)
+target_link_libraries(Actions_test atlascore atlasinsts ${SPARTA} mavis)
```
Comment (colby-nyce) [inline test/actions/CMakeLists.txt]: Can all these be SPARTA_LIBS instead of SPARTA?

**Review by colby-nyce — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (kathlenemagnus) [reply CMakeLists.txt]: Fixed

File: test/actions/CMakeLists.txt (hunk 3 above)
Comment (kathlenemagnus) [reply test/actions/CMakeLists.txt]: Yep, I fixed it!

**Comment by knute-mips** [comment]
Just to put a few thoughts on this ... I realize the use case that's being addressed:  build a local sparta and point Atlas to it.  

When I made the CMake changes a few PRs ago, I intended Atlas to _always_ use an installed version of sparta, even a local one for testing.  So, the idea was if you are making local Sparta changes, you would _locally_ install Sparta and point Atlas to it:
```
# Build sparta/install
cd sparta/release
make
cmake --install .  --prefix $HOME

# Build Atalas
cd atlas/build
cmake -DSPARTA_SEARCH_DIR=$HOME ...
make -j
```

Agreed, it's an annoyance to build/install sparta for every change, but keeps Atlas' build env consistent.  Up to you guys.  :)

**Pull request merged**